workos/resources/sso.py

```diff
@@ -1,87 +1,59 @@
-from workos.resources.base import WorkOSBaseResource
+from typing import List, Literal, Union
 
+from workos.resources.workos_model import WorkOSModel
+from workos.typing.literals import LiteralOrUntyped
+from workos.utils.connection_types import ConnectionType
 
-class WorkOSProfile(WorkOSBaseResource):
-    """Representation of a User Profile as returned by WorkOS through the SSO feature.
 
-    Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSProfile is comprised of.
-    """
+class Profile(WorkOSModel):
+    """Representation of a User Profile as returned by WorkOS through the SSO feature."""
 
-    OBJECT_FIELDS = [
-        "id",
-        "email",
-        "first_name",
-        "last_name",
-        "groups",
-        "organization_id",
-        "connection_id",
-        "connection_type",
-        "idp_id",
-        "raw_attributes",
-    ]
+    object: Literal["profile"]
+    id: str
+    connection_id: str
+    connection_type: LiteralOrUntyped[ConnectionType]
+    organization_id: Union[str, None]
+    email: str
+    first_name: Union[str, None]
+    last_name: Union[str, None]
+    idp_id: str
+    groups: Union[List[str], None]
+    raw_attributes: dict
 
 
-class WorkOSProfileAndToken(WorkOSBaseResource):
-    """Representation of a User Profile and Access Token as returned by WorkOS through the SSO feature.
+class ProfileAndToken(WorkOSModel):
+    """Representation of a User Profile and Access Token as returned by WorkOS through the SSO feature."""
 
-    Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSProfileAndToken is comprised of.
-    """
+    access_token: str
+    profile: Profile
 
-    OBJECT_FIELDS = [
-        "access_token",
-    ]
 
-    @classmethod
-    def construct_from_response(cls, response):
-        profile_and_token = super(WorkOSProfileAndToken, cls).construct_from_response(
-            response
-        )
+ConnectionState = Literal["linked", "unlinked"]
```

Comment:
OK, our [API reference](https://workos.com/docs/reference/sso/connection) claim the valid values are `active`, `inactive`, and `validating`. But looking at the actual [JSON model](https://github.com/workos/workos/blob/9b93b7ad269ec28a4a0f9d024c84475cd44c5406/packages/api-json-models/src/shared/connection-state.ts#L3), we have 6 different states.

I assume some of these states might always be filtered out by the service. But if they're impossible to expose, they should not be in our API JSON model. :/